core/src/main/java/org/mule/internal/connection/CachedConnectionProviderWrapper.java

```diff
@@ -39,5 +60,42 @@ public ConnectionValidationResult validate(Connection connection)
         }
         return getDelegate().validate(connection);
     }
+
+    /**
+     * @return a {@link RetryPolicyTemplate} with the configured values in the Mule Application.
+     */
+    public RetryPolicyTemplate getRetryPolicyTemplate()
+    {
+        return retryPolicyTemplate;
+    }
+
+    public void setMuleContext(MuleContext muleContext)
+    {
+        this.muleContext = muleContext;
+    }
+
+    @Override
+    public void dispose()
+    {
+
+    }
+
+    @Override
+    public void initialise() throws InitialisationException
+    {
+        LifecycleUtils.initialiseIfNeeded(retryPolicyTemplate, true, muleContext);
```

Comment:
propagate lifecycle on all phases. Otherwise, you're speculating with the fact that you will only get a retry policy which only requires initialise()